File: backend/src/main/java/org/example/backend/domain/fish/dto/FishRequestDto.java
package org.example.backend.domain.fish.dto;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

public record FishRequestDto(
    @NotNull
    @Size(min = 1, max = 50)
    String species,

    @NotNull
    @Size(min = 1, max = 50)
    String name
) {

}

File: backend/src/main/java/org/example/backend/domain/fish/dto/FishRequestDto.kt
package org.example.backend.domain.fish.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.Size

data class FishRequestDto(
    @field:NotBlank
    @field:Size(min = 1, max = 50)
    val species: String,

    @field:NotBlank
    @field:Size(min = 1, max = 50)
    val name: String
)
